plugins/net_plugin/net_plugin.cpp

```diff
@@ -1221,12 +1263,11 @@ namespace eosio {
       buffer_queue.fill_out_buffer( bufs );
 
       strand.post( [c{std::move(c)}, bufs{std::move(bufs)}]() {
-         boost::asio::async_write( *c->socket, bufs,
-            boost::asio::bind_executor( c->strand, [c, socket=c->socket]( boost::system::error_code ec, std::size_t w ) {
+         auto write_lambda = [c, psocket=&c->get_socket()]( boost::system::error_code ec, std::size_t w ) {
```

Comment:
This needs to capture the shared_ptr of the socket so it maintains the lifetime of the socket.